util/src/network/host.rs

```diff
@@ -690,26 +689,34 @@ impl<Message> Host<Message> where Message: Send + Sync + Clone {
 			FIRST_HANDSHAKE ... LAST_HANDSHAKE => {
 				let handshakes = self.handshakes.write().unwrap();
 				if let Some(handshake) = handshakes.get(token).cloned() {
-					failure_id = Some(handshake.lock().unwrap().id().clone());
+					let mut handshake = handshake.lock().unwrap();
```

Comment:
Not much happens under this lock `io.deregister_stream()` does not lock anything, just sends a message. So I'm pretty sure that it is safe.